src/libmongoc/CMakeLists.txt

```diff
@@ -708,6 +707,15 @@ set (STATIC_LIBRARIES
    ${SNAPPY_LIBRARIES} ${ZLIB_LIBRARIES} ${MONGOC_ZSTD_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${ICU_LIBRARIES} ${LIBMONGOCRYPT_LIBRARY}
 )
 
+# utf8proc configuration
+find_package(Utf8Proc REQUIRED)
+if ("${USE_BUNDLED_UTF8PROC}" STREQUAL "TRUE")
+   list(APPEND SOURCES $<TARGET_OBJECTS:utf8proc_obj>)
+else()
+   list(APPEND LIBRARIES utf8proc)
+   list(APPEND STATIC_LIBRARIES utf8proc)
```

Comment:
Use the _mongo-dependencies target (defined just below here) to propagate these linker requirements